crates/astria-proto/src/native/sequencer/v1alpha1.rs

```diff
@@ -1251,6 +1266,1158 @@ impl Display for IncorrectAddressLength {
+        let Some(tendermint::Hash::Sha256(data_hash)) = header.data_hash else {
+            return Err(SequencerBlockError::FieldNotSet("header.data_hash"));
+        };
+
+        let rollup_transactions = rollup_transactions
+            .into_iter()
+            .map(rollup_txs_to_tuple)
+            .collect::<Result<_, _>>()
+            .map_err(SequencerBlockError::ParseRollupTransactions)?;
+
+        if !are_rollup_txs_included(&rollup_transactions, &rollup_transactions_proof, data_hash) {
+            return Err(SequencerBlockError::RollupTransactionsNotInSequencerBlock);
+        }
+        if !are_rollup_ids_included(
+            rollup_transactions.keys().copied(),
+            &rollup_ids_proof,
+            data_hash,
+        ) {
+            return Err(SequencerBlockError::RollupIdsNotInSequencerBlock);
+        }
+
+        Ok(Self {
+            block_hash,
+            header,
+            rollup_transactions,
+            rollup_transactions_proof,
+            rollup_ids_proof,
+        })
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("failed constructing a celestia rollup blob")]
+pub struct CelestiaRollupBlobError {
+    #[source]
+    kind: CelestiaRollupBlobErrorKind,
+}
+
+impl CelestiaRollupBlobError {
+    fn field_not_set(field: &'static str) -> Self {
+        Self {
+            kind: CelestiaRollupBlobErrorKind::FieldNotSet {
+                field,
+            },
+        }
+    }
+
+    fn rollup_id(source: IncorrectRollupIdLength) -> Self {
+        Self {
+            kind: CelestiaRollupBlobErrorKind::RollupId {
+                source,
+            },
+        }
+    }
+
+    fn proof(source: <merkle::Proof as Protobuf>::Error) -> Self {
+        Self {
+            kind: CelestiaRollupBlobErrorKind::Proof {
+                source,
+            },
+        }
+    }
+
+    fn sequencer_block_hash(actual_len: usize) -> Self {
+        Self {
+            kind: CelestiaRollupBlobErrorKind::SequencerBlockHash(actual_len),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+enum CelestiaRollupBlobErrorKind {
+    #[error("the expected field in the raw source type was not set: `{field}`")]
+    FieldNotSet { field: &'static str },
+    #[error("failed converting the provided bytes to Rollup ID")]
+    RollupId { source: IncorrectRollupIdLength },
+    #[error("failed constructing a Merkle Hash Tree Proof from the provided raw protobf type")]
+    Proof {
+        source: <merkle::Proof as Protobuf>::Error,
+    },
+    #[error(
+        "the provided bytes were too short for a sequencer block hash. Expected: 32 bytes, \
+         provided: {0}"
+    )]
+    SequencerBlockHash(usize),
+}
+
+/// A shadow of [`CelestiaRollupBlob`] with public access to all its fields.
+///
+/// At the moment there are no invariants upheld by [`CelestiaRollupBlob`] so
+/// they can be converted directly into one another. This can change in the future.
+pub struct UncheckedCelestiaRollupBlob {
```

Comment:
Why the "unchecked" verbiage here? There is nothing checked when converting? It's just pub access vs not?

Reply:
Yeah, thats the case here.

Honestly no reason other than being symmetric with the sequencer blob.

I'm not a big fan of the unchecked versions. But yes, it's to give pub access to the fields to avoid allocations.